SIPS/sip-20.md

```diff
@@ -139,11 +169,37 @@ type WebSocketTextMessage = {
  * @property type - The type of the message.
  * @property message - The message as an ArrayBuffer.
  */
-type WebSocketBinaryMessage = {
+type WebSocketIncomingBinaryMessage = {
   type: 'binary';
   message: ArrayBuffer;
 };
 
+/**
+ * A text message sent to a WebSocket.
+ *
+ * @property type - The type of the message.
+ * @property message - The message as a string.
+ * @property destination - The destination web socket URL as a string.
+ */
+type WebSocketOutgoingTextMessage = {
+    type: 'text';
+    message: string;
+    destination: string;
+};
+
+/**
+ * A binary message sent to a WebSocket.
+ *
+ * @property type - The type of the message.
+ * @property message - The message as an ArrayBuffer.
+ * @property destination - The destination web socket URL as a string.
+ */
+type WebSocketOutgoingBinaryMessage = {
+    type: 'binary';
+    message: ArrayBuffer;
+    destination: string;
+};
```

Comment:
This can be simplified:

```ts
type WebSocketIncomingMessage = WebSocketIncomingTextMessage | WebSocketIncomingBinaryMessage;
type WebSocketOutgoingMessage = WebSocketIncomingMessage & {
  destination: string;
};
```